src/bootstrap/config.rs

```diff
@@ -912,10 +910,14 @@ impl Config {
             config.out = crate::util::absolute(&config.out);
         }
 
-        if !has_custom_rustc && !config.initial_rustc.starts_with(&config.out) {
```

Comment:
I don't really understand what this check was trying to accomplish -- it seems pretty odd to me that config.out matters here. The new logic just trusts build.rustc or build.cargo, and falls back to the stage0 path as before.

Reply:
I don't quite understand why this behavior is different than before; don't we set `initial_rustc` on line 906 above? I think the bug is that I forgot the same check for `initial_cargo`.

That said, I agree the new logic is simpler and easier to understand, so :+1: for using it for initial_rustc as well.

> I don't really understand what this check was trying to accomplish -- it seems pretty odd to me that config.out matters

I was trying to be conservative and only change the behavior when bootstrap was downloaded from CI (i.e. `config.out` wasn't a parent of the `RUSTC` env variable bootstrap was built with). But I think your approach is fine too and easier to maintain - ideally we wouldn't set `RUSTC` in the build script at all.

Reply:
I don't think the build-script set RUSTC matters at all at this point? It's not getting read by config.rs anyway, so I would hope it doesn't :)

Reply:
Oh you're right, I'd forgotten I'd completely removed that: https://github.com/rust-lang/rust/pull/98483/files#diff-7eddc76f1be9eca2599a9ae58c65ffe247fbdff9b02ef687439894cab9afe749L781

Yes, this is absolutely the right fix then if there's no default for `initial_rustc`.